tests/mock-snapd.c

```diff
@@ -20,6 +20,53 @@
 
 #include "mock-snapd.h"
 
+#if !GLIB_CHECK_VERSION(2, 66, 0)
+
+#define HEX_CHAR_TO_NUMBER(c) (((c) < 'a') ? (c) - '0' : ((c) - 'a' + 10))
+
+// a quick&dirty replacement to allow the tests to work, since g_uri_parse_params
+// wasn't added until GLib 2.66
+
+// Replaces in-place any %XY token with the corresponding 0xXY byte
+static void parse_string(gchar *str) {
```

Comment:
This could probably do with a better name.

Reply:
You are right. Changed to `decode_url_escape_codes()`.